src/configure

```diff
@@ -251,14 +310,29 @@ function check_library {
 }
 
 
-
 #Check if at least one of these variables is set
 #If yes, we want to switch to using the MKL
 is_set $MKLLIBDIR && echo "Force-configuring KALDI to use MKL" && export MATHLIB="MKL"
 is_set $MKLROOT && echo "Force-configuring KALDI to use MKL"&& export MATHLIB="MKL"
 is_set $CLAPACKROOT && echo "Force-configuring KALDI to use CLAPACK"&& export MATHLIB="CLAPACK"
 is_set $OPENBLASROOT && echo "Force-configuring KALDI to use OPENBLAS"&& export MATHLIB="OPENBLAS"
 
+
+# If HOST is specified, parse the TARGET_ARCH, otherwise use uname -m
+if [[ "$HOST" == "" ]] ; then
+  TARGET_ARCH="`uname -m`"
+else
+  IFS='-' read -ra PARTS <<< "$HOST"
```

Comment:
please indicate what's going on here using an example, this code is quite complicated.